### PR TITLE
add support for serverless-offline's cache invalidation

### PR DESCRIPTION
**Pull request by DanielSchaffer**
`serverless-offline`, by default, invalidates the node.js require cache each time `createHandler` is called. This allows changes to the handler js files to be picked up automatically during development. This feature is on by default, and is disabled by setting the `skipCacheInvalidation` config to `true`. This PR adds support for that same functionality.

Normally, I'd argue that this would be a breaking change, since it's changing the default functionality, and users would have to explicitly opt out in order to retain the original functionality. However, in this case, this functionality is already the default in the host project (`serverless-offline`), and the expectation for me at least is that plugins would follow the same functionality.

I've also slightly augmented the functionality such that an array of RegExp patterns can be passed as the `skipCacheInvalidation` property, which will allow certain files or directories to be excluded from cache invalidation. This was actually necessary in order to make the unit test work, but I think it's also a nice functional addition, and I'll be submitting a PR to the `serverless-offline` project so that it's supported there as well.

### CODE REVIEW

**Comment by DanielSchaffer** [comment]
This one's going to conflict with the other one ... let me update my branch on my end and I'll repush

**Comment by mj1618** [comment]
Yea sure no problem

File: test/spec/sns.ts [hunk 1]
```diff
@@ -204,9 +219,53 @@ const createServerless = (accountId: number, handlerName: string = "pongHandler"
     };
 };
 
+const createServerlessCacheInvalidation = (accountId: number, handlerName: string = "pongHandler", host: string = null) => {
+    return {
+        config: {
+            skipCacheInvalidation: [
+                /mock\.state/,
```
Comment (mj1618) [inline test/spec/sns.ts]: Cool!

File: src/index.ts [hunk 2]
```diff
@@ -158,6 +158,26 @@ class ServerlessOfflineSns {
...
+
+                // if an array is provided to the serverless config, check the entries there too
+                if (Array.isArray(this.serverless.config.skipCacheInvalidation) &&
+                    this.serverless.config.skipCacheInvalidation.find(pattern => new RegExp(pattern).test(key))) {
+                    continue;
+                }
+
+                delete require.cache[key];
+            }
+        }
```
Comment (mj1618) [inline src/index.ts]: This makes sense, good comments.

**Comment by DanielSchaffer** [comment]
Okay, should be good to go now, tests are passing.

**Comment by mj1618** [comment]
You should've rebased/merged with master I think, I still see conflicts

**Comment by DanielSchaffer** [comment]
Yeah, I thought I did, but I forgot to pull from the original repo. Should be all fixed up now.

**Pull request merged**